launch_pytest/launch_pytest/tools/process.py

```diff
@@ -124,6 +104,24 @@ async def wait_for_output(
         timeout)
 
 
+async def assert_output(
+    launch_context, execute_process_action, validate_output, timeout=None
+):
+    def condition():
+        try:
+            validate_output(execute_process_action.get_stdout())
+        except AssertionError:
+            return False
+        return True
+    cond_value = await _wait_for_event_with_condition(
+        launch_context,
+        execute_process_action,
+        _get_stdout_event_handler,
+        condition,
+        timeout)
+    return cond_value if cond_value else validate_output(execute_process_action.get_stdout())
```

Comment:
@ivanpauno nit: why evaluate the predicate one last time (as opposed to just returning what comes out of `await`)?

Reply:
to actually throw an assertion error to the user, validate_output() is being wrapped in a condition that catches the AssertionError

Reply:
Ah good point. Why do we `return` though?

Reply:
ah yeah, no need of returning anything here

Reply:
Fixed in https://github.com/ros2/launch/pull/553/commits/f1abf9fd6b52bbf2405d0bb69b674aa4f80ae8b4.